### DISCUSSION TITLE
Endpoint is blocking the entire server after 6 requests

### DISCUSSION THREAD

**Discussion by saif-ellafi** [fastapi/fastapi#6540] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
app = FastAPI()
app.mount("/static", StaticFiles(directory="static"), name="static")

@app.get("/", response_class=HTMLResponse)
def home(request: Request):
    return templates.TemplateResponse("forecast_app.html", {"request": request})

def run_on_aws_and_wait():
    pass # stuff getting done remotely

@app.get('/submit_forecast')
def submit_forecast():
    return {'data': run_on_aws_and_wait()}
```

### Description

I have an endpoint on FastAPI that runs some code in a (AWS) lambda function, which takes about 2 minutes to finish.
The desired behaviour is to be able to run at least 100 of these simultaneously.

After making 6 requests to this endpoint, the entire webserver becomes unresponsive (including the entrypoint `/`), together with all other endpoints or anything that I intend to do with it, until one of these requests finishes. This task is not CPU intensive, as it runs remotely, it is mostly waiting. For longer endpoints I do respond with an ACCEPTED and pool it later, but this endpoint I need to wait for the response.

I have read through the entire documentation (https://fastapi.tiangolo.com/async), and many other similar issues (like this one https://github.com/tiangolo/fastapi/issues/2484), which lead me to test things like:
1. making the function async together with a `run_in_threadpool` awaiting call. No success.
2. increase the number of workers. No success.
3. Tried both asyncio loops and uvloops. No success.
4. Running the server through gunicorn doesn't help.

I am running this application with `uvicorn main:app --port 5000 --host 0.0.0.0 --workers 20 --limit-concurrency 100`

Also read hundreds of other similar issues but all end by being solved with either nested subprocesses (which is more or less the same as having a standard `def` function or throughout asyncio calls). But honestly none of them solve it for me. I also can't figure where the number 6 comes from, my hypothesis is that of course it is related to the ThreadPoolExecutor max_workers, but I am not sure really.

Any help welcome

UPDATE 1: I just realized this 6 request block is per browser? Opening an incognito window allows for more requests.

### Operating System

Linux

### FastAPI Version

0.65.2

### Python Version

3.7.3

### Additional Context

4 Cores Virtual Machine

**Comment by haviduck** [comment]
does it happen if you set env WEB_CONCURRENCY=8?

**Comment by ycd** [comment]
> UPDATE 1: I just realized this 6 request block is per browser? Opening an incognito window allows for more requests

There was a similar issue which was blocking the requests after 7th tab (see: https://github.com/tiangolo/fastapi/issues/2786), it is some kind of limitation of the browser, your server should be fine with proper benchmarking tool (see: [hey](https://github.com/rakyll/hey))

**Comment by saif-ellafi** [comment, marked as answer]
Thank you, this was indeed a browser block.
